### DISCUSSION TITLE
ping not working on WSL 2/openSUSE Tumbleweed

### DISCUSSION THREAD

**Discussion by zhang-stephen** [microsoft/WSL#8484] [General] [answered]
### Version

Windows 10 Enterprise 20H2

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.16.3-microsoft-standard-WSL2

### Distro Version

openSuse Tumbleweed

### Repro Steps

1. install openSUSE Tumbleweed from Microsoft Store as WSL 1
2. update it to WSL 2, reboot PC
3. run `ping github.com` and nothing outputs to screen. 

### Expected Behavior

some output to screen, such as the returns of ICMP or error message.

### Actual Behavior

![image](https://user-images.githubusercontent.com/17326718/172185857-b849523b-a97a-4b6e-a2de-d0c6fb8d2660.png)
I got nothing... I have found issue #5109, but it is not for openSUSE

**Comment by elsaco** [comment]
@stark-zhang try pinging as root:

As regular user:
`tux@RIPPER:~> ping github.com`

Using sudo:
```
tux@RIPPER:~> sudo ping github.com
PING github.com (192.30.255.112) 56(84) bytes of data.
64 bytes from lb-192-30-255-112-sea.github.com (192.30.255.112): icmp_seq=1 ttl=49 time=13.5 ms
```

**Comment by sbradnick** [comment]
Both ways mentioned in #5109 make things work, regardless of being Debian/Ubuntu/openSUSE.

You can add the following to `%HOMEPATH%/.wslconfig`
```
[wsl2]
kernelCommandLine=sysctl.net.ipv4.ping_group_range=\"0 2147483647\"
```

or install `libcap-progs` and run: `sudo setcap cap_net_raw+ep /usr/bin/ping` (~~Ubuntu does this~~ Ubuntu uses this method vs. adjusting net.ipv4.ping_group_range).

**Comment by elsaco** [comment]
Looking at `Arch` and `Thumbleweed` both have the same `net.ipv4.ping_group_range` setting:

```
sysctl net.ipv4.ping_group_range
net.ipv4.ping_group_range = 1   0
```

On `Thumbleweed` there's no `libcap` package available:

```
tux@RIPPER:~> sudo zypper in libcap
[sudo] password for root:
Loading repository data...
Reading installed packages...
'libcap' not found in package names. Trying capabilities.
No provider of 'libcap' found.
Resolving package dependencies...
Nothing to do.
```

However, `libcap2` and `libcap-ng0` are installed already. `libcap2` is supposed to offer the same functionality as `libcap` on `Arch`. They are both version `2.64`

Yet on `Thumbleweed` this is the output when running `ping` as non-privileged user:

```
tux@RIPPER:~> ping -v github.com
ping: socket: Operation not permitted
ping: socket: Operation not permitted
```

What other `sysctl` setting must be changed for `ping` to work as it does on `Arch`. I even copied the `ping` binary from `Arch` to `Thumbleweed` and the  output is the same.

**Comment by sbradnick** [comment]
The package which provides getcap/setcap is called `libcap-progs`.

There are [at least] 3 options:

1. `sudo setcap cap_net_raw+ep /usr/bin/ping`
2. Add your own `/etc/sysctl.d/99-my-ping_group_range-fix.conf` with `net.ipv4.ping_group_range = 0 2147483647`
3. edit/create `%HOMEPATH%/.wslconfig` and add the `kernelCommandLine=sysctl.net.ipv4.ping_group_range=\"0 2147483647\"` line. Make sure to do a `wsl --shutdown` before checking again.

**Comment by elsaco** [comment, marked as answer]
Thank you @sbradnick . After installing `libcap-progs` it works w/out `sudo`:

```
tux@RIPPER:~> sudo setcap cap_net_raw=p /usr/bin/ping
[sudo] password for root:
tux@RIPPER:~> getcap /usr/bin/ping
/usr/bin/ping cap_net_raw=p
tux@RIPPER:~> ping github.com
PING github.com (192.30.255.113) 56(84) bytes of data.
64 bytes from lb-192-30-255-113-sea.github.com (192.30.255.113): icmp_seq=1 ttl=49 time=15.9 ms
```

**Comment by ghost** [comment]
Converting to discussion since it appears that the issue has been resolved

**Comment by sbradnick** [comment]
Just wanted to add that ping (via `net.ipv4.ping_group_range`) works out-of-the-box when using systemd in WSLv2. This was surely always the case, setup wise, but the introduction of systemd in WSLv2 put it all together.

in `/etc/wsl.conf`:
```
# export SYSTEMD_UTIL_DIR=$(rpm -E '%{_systemd_util_dir}')
# ensure 'ln -s $SYSTEMD_UTIL_DIR/systemd /sbin/init' or install 'wsl_systemd' pattern
[boot]
systemd=true
```
exit the distsro. Then run `wsl --shutdown` (or wait for it to self-close), next boot of the distro should use systemd.

The options outlined above should work fine for anyone staying on WSLv1 or not enabling systemd in WSLv2.
